state_machine/plugs/flight/lib/pycubed.py

```diff
@@ -147,289 +147,467 @@ def __init__(self):
             print(f'[ERROR][IMU] {e}\n\tMaybe try address=0x68?')
 
         # Initialize Sun Sensors
+        sun_sensors = []
+
         try:
             sun_yn = adafruit_tsl2561.TSL2561(self.i2c2, address=0x29)  # -Y
+            sun_sensors.append(sun_yn)
+        except Exception as e:
+            print('[ERROR][Sun Sensor -Y]', e)
+            
+        try:
             sun_zn = adafruit_tsl2561.TSL2561(self.i2c2, address=0x39)  # -Z
+            sun_sensors.append(sun_zn)
+        except Exception as e:
+            print('[ERROR][Sun Sensor -Z]', e)
+            
+        try:
             sun_xn = adafruit_tsl2561.TSL2561(self.i2c1, address=0x49)  # -X
-
+            sun_sensors.append(sun_xn)
+        except Exception as e:
+            print('[ERROR][Sun Sensor -X]', e)
+            
+        try:
             sun_yp = adafruit_tsl2561.TSL2561(self.i2c1, address=0x29)  # +Y
+            sun_sensors.append(sun_yp)
+        except Exception as e:
+            print('[ERROR][Sun Sensor +Y]', e)
+            
+        try:
             sun_zp = adafruit_tsl2561.TSL2561(self.i2c1, address=0x39)  # +Z
+            sun_sensors.append(sun_zp)
+        except Exception as e:
+            print('[ERROR][Sun Sensor +Z]', e)
+            
+        try:
             sun_xp = adafruit_tsl2561.TSL2561(self.i2c2, address=0x49)  # +X
-            sun_sensors = [sun_zn, sun_yn, sun_xn, sun_yp, sun_xp]
-            for i in sun_sensors:
-                i.enabled = False
-            self.hardware['Sun'] = True
+            sun_sensors.append(sun_xp)
         except Exception as e:
-            print('[ERROR][Sun Sensors]', e)
+            print('[ERROR][Sun Sensor +X]', e)
+            
+        # count the number of sun sensors that we have initialized
+        sun_sensor_count = 0
+        for i in sun_sensors:
+            sun_sensor_count += 1
+            i.enabled = False  # set enabled status to False
+
+        # If there is at least one sun sensor, set to True
+        if sun_sensor_count >= 1:
+            self.hardware['Sun'] = True
 
         # Initialize H-Bridges
+        coils = []
+
         try:
             drv_x = drv8830.DRV8830(self.i2c3, 0x68)  # U6
+            coils.append(drv_x)
+        except Exception as e:
+            print('[ERROR][H-Bridge U6]', e)
+            
+        try:
             drv_y = drv8830.DRV8830(self.i2c3, 0x60)  # U8
+            coils.append(drv_y)
+        except Exception as e:
+            print('[ERROR][H-Bridge U8]', e)
+            
```

Comment:
Similar situation here with code duplication as with the sun sensors

Reply:
Same as above - keep it as is for now, clean it up when we change the attitude control code.